src/features/link/components/links-menu.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  TbBrandFacebook,
+  TbBrandInstagram,
+  TbBrandTiktok,
+  TbBrandTwitter,
+  TbBrandYoutube,
+  TbDotsCircleHorizontal,
+  TbLink,
+} from "react-icons/tb";
+import { match } from "ts-pattern";
+
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+} from "@/features/list";
+import { LinkCategory } from "@/types/enums";
+
+type Link = {
+  url: string;
+  category: LinkCategory;
+};
+
+const LINKS_DATA: Link[] = [
+  {
+    url: "https://www.youtube.com/",
+    category: "youtube",
+  },
+  {
+    url: "https://www.instagram.com/",
+    category: "instagram",
+  },
+  {
+    url: "https://www.tiktok.com/",
+    category: "tiktok",
+  },
+  {
+    url: "https://twitter.com/home",
+    category: "twitter",
+  },
+  {
+    url: "https://www.facebook.com/",
+    category: "facebook",
+  },
+  {
+    url: "https://www.google.com/",
+    category: "hp",
+  },
+  {
+    url: "https://www.google.com/",
+    category: "other",
+  },
+];
+
+const orderedCategories: LinkCategory[] = ["youtube", "instagram", "tiktok", "twitter", "facebook"];
+
+export const LinksMenu = () => {
+  const socialMediaLinks = LINKS_DATA.filter((data) => data.category !== "other" && data.category !== "hp");
```

Comment:
すみません、以前提案したコードについて、type predicateが不要になるので以下の方がよさそうでした。
```suggestion
const orderedCategories = ["youtube", "instagram", "tiktok", "twitter", "facebook"] satisfies LinkCategory[];

export const LinksMenu = () => {
  const socialMediaLinks = LINKS_DATA.flatMap((link) => {
    const category = orderedCategories.find((category) => category === link.category);
    return category ? [{ ...link, category }] : [];
  });
```